### ISSUE TITLE
bug: YouTube Patch Error (The source file doesn't exist)

### ISSUE THREAD

**Issue by HeyItsJono** [revanced/revanced-manager#394]
### Type

Error while running the manager

### Bug description

Trying to patch YouTube fails and raises an error.

### Steps to reproduce

1. Installed YouTube (Version 17.36.37, full APK, tested working).
2. Open Revanced Manager to patch (Root via Magisk).
3. Select desired patches, list at end.
4. Hit patch.
5. Error gets raised.

### Android version

12.1

### Manager version

Version 0.0.32

### Target package name

com.google.android.youtube

### Target package version.

com.google.android.youtube_17.36.37-1531571648_minAPI26(arm64-v8a,armeabi-v7a,x86,x86_64)(nodpi) from apkmirror.com

### Installation type

Root

### Patches selected.

* Client Spoof
* Custom Branding.
* Custom Playback Speed
* Custom Video Buffer
* All of the "Disable X" patches.
* Downloads.
* General/Video Ads.
* All of the "Hide X" patches except hide Cast/Time/Seekbar/Video Buttons.
* Minimised Playback.
* Premium Header.
* Remember Video Quality
* Return YouTube Dislike
* Seekbar Tapping
* sponsorblock.
* swipe controls.
* theme

### Device logs (exported using Manager settings).

```shell
--------- beginning of main
10-14 08:57:55.827 32712  2388 I OpenGLRenderer: Davey! duration=9109ms; Flags=1, FrameTimelineVsyncId=405365, IntendedVsync=3823396434652, Vsync=3823396434652, InputEventId=335735684, HandleInputStart=3823411786348, AnimationStart=3823411789473, PerformTraversalsStart=3823411791713, DrawStart=3823440533434, FrameDeadline=3823429601318, FrameInterval=3823411781765, FrameStartTime=16666666, SyncQueued=3823440735309, SyncStart=3823440856976, IssueDrawCommandsStart=3823440971507, SwapBuffers=3823446586091, FrameCompleted=3832506170327, DequeueBufferDuration=3702865, QueueBufferDuration=954739, GpuCompleted=3832506170327, SwapBuffersCompleted=3823448177185, DisplayPresentTime=1160506089, 
10-14 08:58:09.973 32712  1127 I OpenGLRenderer: Davey! duration=23257ms; Flags=1, FrameTimelineVsyncId=405365, IntendedVsync=3823396434652, Vsync=3823396434652, InputEventId=335735684, HandleInputStart=3823411786348, AnimationStart=3823411789473, PerformTraversalsStart=3823411791713, DrawStart=3823440533434, FrameDeadline=3823446267984, FrameInterval=3823411781765, FrameStartTime=16666666, SyncQueued=3823440735309, SyncStart=3823440856976, IssueDrawCommandsStart=3823440971507, SwapBuffers=3823446586091, FrameCompleted=3846654378559, DequeueBufferDuration=3702865, QueueBufferDuration=954739, GpuCompleted=3846654378559, SwapBuffersCompleted=3823448177185, DisplayPresentTime=1160506089, 
10-14 08:59:11.818 32712  1127 I OpenGLRenderer: Davey! duration=85101ms; Flags=1, FrameTimelineVsyncId=405365, IntendedVsync=3823396434652, Vsync=3823396434652, InputEventId=335735684, HandleInputStart=3823411786348, AnimationStart=3823411789473, PerformTraversalsStart=3823411791713, DrawStart=3823440533434, FrameDeadline=3823462934650, FrameInterval=3823411781765, FrameStartTime=16666666, SyncQueued=3823440735309, SyncStart=3823440856976, IssueDrawCommandsStart=3823440971507, SwapBuffers=3823446586091, FrameCompleted=3908498162845, DequeueBufferDuration=3702865, QueueBufferDuration=954739, GpuCompleted=3908498162845, SwapBuffersCompleted=3823448177185, DisplayPresentTime=1160506089, 
10-14 08:59:32.765 32712 32759 I OpenGLRenderer: Davey! duration=96969ms; Flags=1, FrameTimelineVsyncId=406348, IntendedVsync=3832483667848, Vsync=3832483667848, InputEventId=194415377, HandleInputStart=3832483990221, AnimationStart=3832483992721, PerformTraversalsStart=3832483994648, DrawStart=3832503457097, FrameDeadline=3832516834514, FrameInterval=3832483986887, FrameStartTime=16666666, SyncQueued=3832503540014, SyncStart=3832503587566, IssueDrawCommandsStart=3832503634754, SwapBuffers=3832504346785, FrameCompleted=3929453677380, DequeueBufferDuration=325313, QueueBufferDuration=781771, GpuCompleted=3929453677380, SwapBuffersCompleted=3832505378244, DisplayPresentTime=4722024210397396992, 
10-14 09:00:16.969 32712  1127 I OpenGLRenderer: Davey! duration=127012ms; Flags=1, FrameTimelineVsyncId=407759, IntendedVsync=3846632706425, Vsync=3846632706425, InputEventId=1277880759, HandleInputStart=3846633207672, AnimationStart=3846633209599, PerformTraversalsStart=3846633211109, DrawStart=3846647811111, FrameDeadline=3846665873091, FrameInterval=3846633204495, FrameStartTime=16666666, SyncQueued=3846647931579, SyncStart=3846648025017, IssueDrawCommandsStart=3846648129288, SwapBuffers=3846652262153, FrameCompleted=3973645168554, DequeueBufferDuration=3376094, QueueBufferDuration=786666, GpuCompleted=3973645168554, SwapBuffersCompleted=3846653279236, DisplayPresentTime=1160680169, 
10-14 09:00:58.079 32712  1127 I OpenGLRenderer: Davey! duration=106317ms; Flags=1, FrameTimelineVsyncId=413031, IntendedVsync=3908442566816, Vsync=3908459233482, InputEventId=111999527, HandleInputStart=3908470411905, AnimationStart=3908470413728, PerformTraversalsStart=3908470414405, DrawStart=3908484118052, FrameDeadline=3908475733482, FrameInterval=3908470408155, FrameStartTime=16666666, SyncQueued=3908484247792, SyncStart=3908484352479, IssueDrawCommandsStart=3908484492531, SwapBuffers=3908492430397, FrameCompleted=4014759978587, DequeueBufferDuration=6041199, QueueBufferDuration=1663594, GpuCompleted=4014759978587, SwapBuffersCompleted=3908499177637, DisplayPresentTime=-5476376632407194304, 
10-14 09:01:55.630 32712  2388 I OpenGLRenderer: Davey! duration=142888ms; Flags=1, FrameTimelineVsyncId=415093, IntendedVsync=3929424384606, Vsync=3929424384606, InputEventId=697091033, HandleInputStart=3929424591076, AnimationStart=3929424592169, PerformTraversalsStart=3929424592898, DrawStart=3929447156599, FrameDeadline=3929457551272, FrameInterval=3929424589044, FrameStartTime=16666666, SyncQueued=3929447264307, SyncStart=3929447362588, IssueDrawCommandsStart=3929447463838, SwapBuffers=3929452511755, FrameCompleted=4072313400987, DequeueBufferDuration=4405938, QueueBufferDuration=664635, GpuCompleted=4072313400987, SwapBuffersCompleted=3929453445766, DisplayPresentTime=-86101209380856, 
10-14 09:02:11.469  2418  2418 W sh      : type=1400 audit(0.0:3253): avc: denied { read } for name="/" dev="dm-1" ino=2 scontext=u:r:untrusted_app:s0:c63,c256,c512,c768 tcontext=u:object_r:rootfs:s0 tclass=dir permissive=0 app=app.revanced.manager.flutter
10-14 09:02:11.479  2418  2418 W sh      : type=1400 audit(0.0:3254): avc: denied { read } for name="/" dev="dm-1" ino=2 scontext=u:r:untrusted_app:s0:c63,c256,c512,c768 tcontext=u:object_r:rootfs:s0 tclass=dir permissive=0 app=app.revanced.manager.flutter
10-14 09:02:11.486  2418  2418 W sh      : type=1400 audit(0.0:3255): avc: denied { read } for name="/" dev="dm-1" ino=2 scontext=u:r:untrusted_app:s0:c63,c256,c512,c768 tcontext=u:object_r:rootfs:s0 tclass=dir permissive=0 app=app.revanced.manager.flutter
10-14 09:02:11.496  2418  2418 W sh      : type=1400 audit(0.0:3256): avc: denied { read } for name="/" dev="dm-1" ino=2 scontext=u:r:untrusted_app:s0:c63,c256,c512,c768 tcontext=u:object_r:rootfs:s0 tclass=dir permissive=0 app=app.revanced.manager.flutter
10-14 09:02:11.502  2418  2418 W sh      : type=1400 audit(0.0:3257): avc: denied { read } for name="/" dev="dm-1" ino=2 scontext=u:r:untrusted_app:s0:c63,c256,c512,c768 tcontext=u:object_r:rootfs:s0 tclass=dir permissive=0 app=app.revanced.manager.flutter
10-14 09:02:11.789 32712  5425 I manager.flutte: Waiting for a blocking GC ClassLinker
10-14 09:02:16.357 32712 32712 E BLASTBufferQueue: [ViewRootImpl[MainActivity]#24](f:0,a:2) Applying pending transactions on dtor 1
10-14 09:02:59.511 32712  1127 I OpenGLRenderer: Davey! duration=121465ms; Flags=1, FrameTimelineVsyncId=422424, IntendedVsync=4014736962578, Vsync=4014736962578, InputEventId=332035269, HandleInputStart=4014737353272, AnimationStart=4014737354366, PerformTraversalsStart=4014737355303, DrawStart=4014753895930, FrameDeadline=4014770129244, FrameInterval=4014737351085, FrameStartTime=16666666, SyncQueued=4014754046295, SyncStart=4014754173170, IssueDrawCommandsStart=4014754341034, SwapBuffers=4014757298066, FrameCompleted=4136203070320, DequeueBufferDuration=1792552, QueueBufferDuration=1051354, GpuCompleted=4136203070320, SwapBuffersCompleted=4014758703743, DisplayPresentTime=3,
```

### Installer logs (exported using Installer menu option) [unneeded if issue is not during patching].

```shell
Initializing installer
Creating working directory
Copying original apk
An error occurred! Aborting
Error:
kotlin.io.NoSuchFileException: /data/local/tmp/revanced-manager/com.google.android.youtube/original.apk: The source file doesn't exist.
	at kotlin.io.FilesKt__UtilsKt.copyTo(Utils.kt:219)
	at kotlin.io.FilesKt__UtilsKt.copyTo$default(Utils.kt:217)
	at app.revanced.manager.flutter.MainActivity.runPatcher$lambda-19(MainActivity.kt:126)
	at app.revanced.manager.flutter.MainActivity.$r8$lambda$aEBH7QyMEVbzs_zgD8iozv3hwbI(Unknown Source:0)
	at app.revanced.manager.flutter.MainActivity$$ExternalSyntheticLambda4.run(Unknown Source:24)
	at java.lang.Thread.run(Thread.java:1012)
```

### Acknowledgements

- [X] I have searched the existing issues and this is a new and no duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] I filled out all of the requested information in this issue properly.
- [X] The issue is related solely to the ReVanced Manager

**Comment by Aunali321** [comment]
Try clearing /data/local/tmp/revanced-manager.

**Comment by HeyItsJono** [comment]
> Try clearing /data/local/tmp/revanced-manager.

Thank you, this fixed it in the end. There was a folder `/data/local/tmp/revanced-manager/com.google.android.youtube` with a file `base.apk` inside of it.

Documenting my experience because deleting it was quite difficult, I was unable to do so through root file explorers like MiX or Solid. Navigating there with Termux and attempting to `sudo rm -rf /data/local/tmp/revanced-manager/com.google.android.youtube` gave me an error along the lines of  `base.apk: device or resource busy`.

`lsof +D` showed no useful info. `ps -ef | grep com.google.android.youtube` showed a process locking the folder, but the PID changed constantly and so I was unable to kill the process.

`mount | grep com.google.android.youtube` finally showed that the folder and base.apk were mounted at /dev/block-30 or something along those lines. A single `sudo umount -lf /dev/block-30` didn't do the trick, the folder was still locked for deletion. Running `sudo umount -lf /dev/block-30` about 6 or 7 times finally unmounted the folder and I was able to delete it. 

After deleting this folder, I was able to use Revanced Manager to patch and install youtube without issue.
